projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -67,13 +67,15 @@ export default function PublicizeForm( {
 	const hasInstagramConnection = connections.some(
 		connection => connection.service_name === 'instagram-business'
 	);
-	const [ shouldShowInstagramNotice, setShouldShowInstagramNotice ] = useState(
-		! hasInstagramConnection && isInstagramConnectionSupported
-	);
+
+	const shouldShowInstagramNotice =
+		! hasInstagramConnection &&
+		isInstagramConnectionSupported &&
+		! dismissedNotices.includes( 'instagram' );
+
 	const onDismissInstagramNotice = useCallback( () => {
 		dismissNotice( 'instagram' );
-		setShouldShowInstagramNotice( false );
-	}, [ dismissNotice, setShouldShowInstagramNotice ] );
+	}, [ dismissNotice ] );
```

Comment:
> I was wondering if `useDismissNotice` could take a notice name

That API is not flexible given the rules of React hooks. Sometimes you may want to dismiss multiple notices by passing the value at call time (to the callback)

> so that the `dismissNotice` callback returned wouldn't need to be wrapped in a `useCallback` like this.

That doesn't affect whether the callback should be wrapped in `useCallback`. It will still need to be used.